signer/src/metrics.rs

```diff
@@ -71,6 +73,81 @@ impl Metrics {
         )
         .increment(1);
     }
+
+    /// Increment number of presign requests that were processed noting
+    /// whether the presign validation finished successfully. Also record
+    /// the amount of time that it took to run the validation.
+    ///
+    /// # Notes
+    ///
+    /// If the presign validation result is Ok, just means that the request
+    /// itself included deposits and withdrawals that we know about and
+    /// that there were no duplicates with the request. There could still
+    /// be deposits and withdrawals that failed validation.
+    pub fn increment_presign_validation(instant: Instant, presign_result: &Result<(), Error>) {
```

Comment:
Wouldn't it be better to accept a `Duration` instead of `Instant` (here and below), and require the caller to pass `.elapsed()`?

Reply:
Yeah, that does feel better, will update.